### DISCUSSION TITLE
Support inequality operator and regex for searching alert labels

### DISCUSSION THREAD

**Discussion by gtirloni** [grafana/grafana#73523] [Alerting]
I can use `label:k=v` to search for alerts with label `k` and value `value but there is no way to search when a label doesn't have a specific value or when the label is missing altogether.

This would be useful to detect global inconsistencies in alert rules.

**Comment by konrad147** [comment]
You can try to use `!=` and `!~` operators
- `label:k!=v` will return all alerts with the label `k` different from `v`
- Similarly, `label:k!~v.*` will return all alerts with label `k` that do not match the regex expression

**Comment by gtirloni** [comment]
That's pretty cool, thanks!

I haven't been able to filter alerts missing a label though. It seems the alert must have the label I'm searching for. Any suggestions?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82574. Feel free to continue the discussion around this there. Thank you!
